src/host/ledger.h

```diff
@@ -804,19 +813,19 @@ namespace asynchost
         std::optional<size_t> max_size = std::nullopt;
         if (max_entries_size.has_value())
         {
-          max_size = max_entries_size.value() - entries.size();
+          max_size = max_entries_size.value() - rr.data.size();
         }
         auto v = f_from->read_entries(idx, to_, max_size);
         if (!v.has_value())
         {
-          return std::nullopt;
+          break;
```

Comment:
This is a subtle but important change. Previously, if we had a single chunk that returned no entries, we'd return no entries for the entire range. For instance, if you requested seqnos 5-10, and 10 pushed this range over the max size allowed:
- If 9 and 10 (and anything before) are in the same file (chunk), then this would respond with the contents of 5-9
- If 10 was at the start of a new file (chunk), then this entire request would return nullopt indicating that _nothing_ could be read, ignoring the 5-9 that were already successfully read!